lib/utils.js

```diff
@@ -178,3 +178,24 @@ exports.isBuffer = function (obj) {
 
     return !!(obj.constructor && obj.constructor.isBuffer && obj.constructor.isBuffer(obj));
 };
+
+exports.combine = function combine(a, b) {
+    // we always use both of these, so, let's calculate them now
+    var firstIsArray  = Array.isArray(a);
+    var secondIsArray = Array.isArray(b);
+
+    // mutate `a` to append `b` and then return it
+    if (firstIsArray) {
+        secondIsArray ? a.push.apply(a, b) : a.push(b);
```

Comment:
It's probably a good idea to merge in `utils.combine` with `concat` first into master, and then have the optimization PR be separate.